src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -232,8 +248,16 @@ impl Executor for IndexScanExecutor {
     }
 
     fn collect_output_counts(&mut self, counts: &mut Vec<i64>) {
-        counts.push(self.count);
-        self.count = 0;
+        if self.cursor > self.counts.len() {
+            return;
+        }
+        counts.append(&mut self.counts[..self.cursor].to_vec());
+        self.counts.drain(0..self.cursor);
+        self.cursor = 0;
```

Comment:
Could we remove `self.cursor`, and just use `Vec::last_mut` when we need to update latest range's count?